source/ios/AdaptiveCards/AdaptiveCards/AdaptiveCards/ACRView.mm

```diff
@@ -561,22 +582,55 @@ - (ACOAdaptiveCard *)card
     return _adaptiveCard;
 }
 
-// notifcation is delivered from main (serial) queue, thus run in the main thread context
+// notification is delivered from main (serial) queue, thus run in the main thread context
 - (void)observeValueForKeyPath:(NSString *)path ofObject:(id)object change:(NSDictionary *)change context:(void *)context
 {
-    if ([path isEqualToString:@"image"])
-    {
-        if(context) {
+    if ([path isEqualToString:@"image"]) {
+        bool observerRemoved = false;
+        if (context) {
+            // image that was loaded
             UIImage *image = [change objectForKey:NSKeyValueChangeNewKey];
+            
             NSNumber *number = [NSNumber numberWithUnsignedLongLong:(unsigned long long)(context)];
             NSString *key = [number stringValue];
-            ACRRegistration *reg = [ACRRegistration getInstance];
+            
             ACOBaseCardElement *baseCardElement = _imageContextMap[key];
-            if(baseCardElement) {
+            if (baseCardElement) {
+                ACRRegistration *reg = [ACRRegistration getInstance];
                 ACRBaseCardElementRenderer<ACRIKVONotificationHandler> *renderer = (ACRBaseCardElementRenderer<ACRIKVONotificationHandler> *)[reg getRenderer:[NSNumber numberWithInt:baseCardElement.type]];
-                if(renderer && [[renderer class] conformsToProtocol:@protocol(ACRIKVONotificationHandler)]) {
+                if (renderer && [[renderer class] conformsToProtocol:@protocol(ACRIKVONotificationHandler)]) {
+                    
+                    // handle background image for column and container that uses resource resolver
+                    if (baseCardElement.type == ACRCardElementType::ACRColumn
+                        || baseCardElement.type == ACRCardElementType::ACRContainer) {
+                        UIImageView *imageView = (UIImageView *)object;
+                        BackgroundImage *backgroundImage = (BackgroundImage *)context;
```

Comment:
this context may point to invalid object when image loading happens much later from the time card is rendered. 

can you create an enum, and create a private property that holds this enum. have context to point to that enum for disambiguation?

also please consider moving if and its statement to configUpdateForUIImageVIew methods for container and column renderer. 

this applies to else if and its statement as well.
you can use the enum value pointed by context to land in this branch, and copy the BackgroundImage to ACRView.

you can leave the logic for unsubscribing the observer.

Reply:
@ let me know if this was fixed appropriately. Want to double check.

---
In reply to: [263085503](https://github.com/Microsoft/AdaptiveCards/pull/2388#discussion_r263085503) [](ancestors = 263085503)